bundle/src/main/java/com/adobe/acs/commons/properties/filter/CapturingResponseWrapper.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2020 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.adobe.acs.commons.properties.filter;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+
+/**
+ * Response Wrapper used to allow for full output of the current response data so that it can be manipulated
+ * and rewritten.
+ */
+public class CapturingResponseWrapper extends HttpServletResponseWrapper {
+
+    private final ByteArrayOutputStream capture;
+    private ServletOutputStream output;
+    private PrintWriter writer;
+
+    public CapturingResponseWrapper(HttpServletResponse response) {
+        super(response);
+        capture = new ByteArrayOutputStream(response.getBufferSize());
+    }
+
+    @Override
+    public ServletOutputStream getOutputStream() {
+        if (writer != null) {
+            throw new IllegalStateException(
+                    "getWriter() has already been called on this response.");
+        }
+
+        if (output == null) {
+            output = new ServletOutputStream() {
```

Comment:
out of curiosity, do we know if this in any way resembles how the HTML rewriter safely does it's thing in AEM?

Reply:
I'm not sure if this is similar to how the HTML transformer pipeline works - i imagine it's not quite the same since this doesn't seem to be chunked in the same way

Reply:
Since this only executes on JSON requests and only if the feature is turned on, and it seems to be working fine for various sizes of JSON I think I feel okay with it (though I think we should still make `isReady` return `true` per the other comment.